docs/admin/cluster-components.md

```diff
@@ -0,0 +1,140 @@
+
+<strong>Documentation for specific releases can be found at
+[releases.k8s.io](http://releases.k8s.io).</strong>
+
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+# Kubernetes Cluster Admin Guide: Cluster Components
+
+This document outlines the various binary components that need to run to
+deliver a functioning Kubernetes cluster.
+
+## Master Components
+
+Master components are those that provide the cluster's control plane. For
+example, master components are responsible for making global decisions about the
+cluster (e.g., scheduling), and detecting and responding to cluster events
+(e.g., starting up a new pod when a replication controller's 'replicas' field is
+unsatisfied).
+
+Master components could in theory be run on any node in the cluster. However,
+for simplicity, current set up scripts typically start all master components on
+the same VM, and does not run user containers on this VM. See
+[availability.md](availability.md) for an example multi-master-VM setup.
+
+Even in the future, when Kubernetes is fully self-hosting, it will probably be
+wise to only allow master components to schedule on a subset of nodes, to limit
+co-running with user-run pods, reducing the possible scope of a
+node-compromising security exploit.
+
+### kube-apiserver
+
+[kube-apiserver](kube-apiserver.md) exposes the Kubernetes API; it is the front-end for the
+Kubernetes control plane. It is designed to scale horizontally (i.e., one scales
+it by running more of them-- [availability.md](availability.md)).
+
+### etcd
+
+```etcd``` is used as Kubernetes' backing store. All cluster data is stored here.
+Proper administration of a Kubernetes cluster includes a backup plan for etcd's
+data.
+
+### kube-controller-manager
+
+[kube-controller-manager](kube-controller-manager.md) is a binary that runs controllers, which are the
+background threads that handle routine tasks in the cluster. Logically, each
+controller is a separate process, but to reduce the number of moving pieces in
+the system, they are all compiled into a single binary and run in a single
+process.
+
+These controllers include:
+
+* Node Controller
+ * Responsible for noticing & responding when nodes go down.
+* Replication Controller
+ * Responsible for maintaining the correct number of pods for every replication
+   controller object in the system.
+* Endpoints Controller
+ * Populates the Endpoints object (i.e., join Services & Pods).
+* Service Account & Token Controllers
+ * Create default accounts and API access tokens for new namespaces.
+* ... and others.
+
+### kube-scheduler
+
+[kube-scheduler](kube-scheduler.md) watches newly created pods that have no node assigned, and
+selects a node for them to run on.
+
+### addons
+
+Addons are pods and services that implement cluster features. They aren't
```

Comment:
I think this paragraph could be rewritten a bit more concisely: "Addons are pods and services that implement cluster features. They don't run on the master VM, but currently the default setup scripts (link stuff) that make the API calls to create these pods and services do run on the master VM."

(the stuff about "run as pods within the cluster, not directly on VMs of the cluster" is confusing and I'd just drop it)

BTW should you say something about the kube-system namespace?